exp-models/addon/utils/csv-writer.js

```diff
@@ -0,0 +1,107 @@
+import Ember from 'ember';
+
+/**
+ * New line string
+ *
+ * @type {string}
+ */
+export const NL = '\r\n';
+
+/**
+ * Flatten a nested object into a single level, with dotted paths for keys
+ *
+ * @param obj {!Object} - The object to flatten
+ * @param prefix {string} - The prefix for the keys
+ * @returns {Object} - The flattened object
+ */
+function squashPrefix(obj, prefix) {
+    let ret = {};
+
+    for (const key in obj) {
+        if (!obj.hasOwnProperty(key)) {
+            continue;
+        }
+
+        const newPrefix = prefix ? `${prefix}.${key}` : key;
+
+        let value = Ember.get(obj, key);
+
+        if (value && value.toJSON) {
+            value = value.toJSON();
+        }
+
+        if (Ember.$.isPlainObject(value)) {
+            Object.assign(ret, squashPrefix(value, newPrefix));
+        } else {
+            ret[newPrefix] = value;
+        }
+    }
+
+    return ret;
+}
+
+/**
+ * Flatten a nested object into a single level, with dotted paths for keys
+ *
+ * @param obj {!Object} - The object to flatten
```

Comment:
My mistake, it looks so similar to JSDoc, I thought it was. Exclamation in JSDoc mean a required parameter. Will change it to YUIDoc syntax.